src/binary-writer.cc

```diff
@@ -312,34 +401,43 @@ Index BinaryWriter::GetEventVarDepth(const Var* var) {
   return var->index();
 }
 
+template <typename T>
+Index BinaryWriter::InternSymbol(const std::string& name, uint8_t flags,
+                                 const T& arg) {
+  auto iter = symtab_.find(name);
+  if (iter != symtab_.end()) {
+    Index sym_index = iter->second;
+    assert(symbols_[sym_index].type() == T::type);
```

Comment:
I'm not sure whether this assert is safe; seems like it could be invalid based on malformed data in wasm module. @sbc100 do you know?

Reply:
Good point; let's print a warning in that case instead of failing.

In any case, the current logic doesn't actually follow the toolchain-conventions invariants; symbols that are neither imported nor exported by a module have no name uniqueness requirements AFAIU.  I will fix the logic in a followup.